### PR TITLE
Multiple light

### PR DESCRIPTION
**Pull request by zcorniere**

### CODE REVIEW

File: libgraphics/include/pivot/graphics/types/Light.hxx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        alignas(4) float radius = 10.0f;
+        alignas(4) float falloff = 1.0f;
+    };
+    static_assert(sizeof(PointLight) % 4 == 0);
+    static_assert(sizeof(PointLight) == (sizeof(float) * 3) + (sizeof(float) * 3) + (sizeof(float) * 4) + 8);
+    /// @endcond
+
+    /// Represent a Directional Light
+    /// @cond
+    struct DirectionalLight {
```
Comment (Ersikan) [inline libgraphics/include/pivot/graphics/types/Light.hxx:30]: Do you not need a rotation here ?

File: libgraphics/shaders/default_lit.frag [hunk 2]
```diff
@@ -25,35 +29,59 @@ struct Material {
...
 
 layout(set = 0, binding = 2) uniform sampler2D texSampler[];
 
-const vec3 lightDirection = vec3(11.0, 16.0, 24.0);
+layout(std140, set = 1, binding = 2) readonly buffer LightBuffer {
+    PointLight pointLightArray[];
+}  omniLight;
+
+layout(std140, set = 1, binding = 3) readonly buffer DirectLight {
+    DirectionalLight pointLightArray[];
```
Comment (Ersikan) [inline libgraphics/shaders/default_lit.frag:61]:
```suggestion
    DirectionalLight directionalLightArray[];
```

File: libgraphics/shaders/default_pbr.frag [hunk 3]
```diff
@@ -25,13 +29,36 @@ struct Material {
...
 
 layout(set = 0, binding = 2) uniform sampler2D texSampler[];
 
-const vec3 lightDirection = vec3(0.0, 1000.0, 0.0);
+layout(std140, set = 1, binding = 2) readonly buffer LightBuffer {
+    PointLight pointLightArray[];
+}  omniLight;
+
+layout(std140, set = 1, binding = 3) readonly buffer DirectLight {
+    DirectionalLight pointLightArray[];
```
Comment (Ersikan) [inline libgraphics/shaders/default_pbr.frag:60]:
```suggestion
    DirectionalLight directionalLightArray[];
```

File: libgraphics/shaders/default_pbr.frag [hunk 4]
```diff
@@ -102,21 +129,21 @@ void main()
...
-    {
-        vec3 L = normalize(lightDirection - fragPosition);
+    for (uint i = 0; i < cameraData.push.pointLightCount; i++) {
+        PointLight light = omniLight.pointLightArray[i];
+        vec3 L = normalize(light.position.xyz - fragPosition);
         vec3 H = normalize(V + L);
-        float distance = length(lightDirection - fragPosition);
+        float distance = length(light.position.xyz - fragPosition);
         float attenuation = 1.0 / (distance * distance);
         /// Replace the value with the color of the light
```
Comment (Ersikan) [inline libgraphics/shaders/default_pbr.frag:145]:
```suggestion
```

**Review by Ersikan — APPROVED** [review]

File: libgraphics/include/pivot/graphics/types/Light.hxx (hunk 1 above)
Comment (zcorniere) [reply libgraphics/include/pivot/graphics/types/Light.hxx:30]: Yes, I actually only need a rotation,

**Pull request merged**